dali/ft/ftslavelib.cpp

```diff
@@ -0,0 +1,135 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "jliball.hpp"
+#include "platform.h"
+#include "jlib.hpp"
+#include "jio.hpp"
+#include "jmutex.hpp"
+#include "jfile.hpp"
+#include "jsocket.hpp"
+#include "jdebug.hpp"
+
+#include "fterror.hpp"
+#include "dadfs.hpp"
+#include "rmtspawn.hpp"
+#include "filecopy.hpp"
+#include "fttransform.ipp"
+#include "daftformat.hpp"
+#include "daftcfg.hpp"
+#include "mptag.hpp"
+#include "ftslavelib.hpp"
+
+
+bool processPullCommand(ISocket * masterSocket, MemoryBuffer & msg)
+{
+    srand((int)get_cycles_now());
```

Comment:
not new code, but I think this should be ftslave only, or deleted.  There is another call in deserializeAction().

I think it is is only used to make sure each slave does not send their progress updates at the same time.  Better would probably be to directly use get_cycles_now since they will almost certainly not match:

    int adjust = get_cycles_now() % updateFrequency - (updateFrequency/2);

Reply:
agree, odd code, get_cycles_now should suffice in both cases.
I'll change.